api/src/main/java/org/wildfly/transformer/ArchiveTransformer.java

```diff
@@ -13,76 +13,68 @@
+public abstract class ArchiveTransformer {
 
-    private ToolUtils() {
-        // forbidden instantiation
-    }
+    protected final Map<Config, String> configs;
+    protected final boolean verbose;
 
-    public static final String CLASS_FILE_EXT = ".class";
-    public static final String JAR_FILE_EXT = ".jar";
+    protected ArchiveTransformer(final Map<Config, String> configs, final boolean verbose) {
+        this.configs = configs;
+        this.verbose = verbose;
+    }
 
-    public static void transformClassFile(final File inClassFile, final File targetDir, final String packagesMappingFile) throws IOException {
-        if (inClassFile.length() > Integer.MAX_VALUE) {
-            throw new UnsupportedOperationException("File " + inClassFile.getAbsolutePath() + " too big! Maximum allowed file size is " + Integer.MAX_VALUE + " bytes");
-        }
-        if (!targetDir.exists()) throw new IOException("Couldn't find directory: " + targetDir.getAbsolutePath());
-        final Transformer t = newTransformer(packagesMappingFile);
-        byte[] clazz = new byte[(int)inClassFile.length()];
-        readBytes(new FileInputStream(inClassFile), clazz, true);
-        final Resource origResource = new Resource(inClassFile.getName(), clazz);
-        final Resource[] newResources = t.transform(origResource);
-        if (newResources.length == 0) {
-            writeBytes(new FileOutputStream(new File(targetDir, origResource.getName())), origResource.getData(), true);
-        } else {
-            for (Resource newResource : newResources) {
-                writeBytes(new FileOutputStream(new File(targetDir, newResource.getName())), newResource.getData(), true);
+    /**
+     * Attempts to apply configured transformations to given <code>source</code> archive and produces new <code>target</code> archive.
+     *
+     * @param inJarFile archive file to be consumed (can be exploded)
+     * @param outJarFile archive file to be produced (will be exploded if source was exploded)
+     * @return <code>true</code> if transformations were applied to archive, <code>false</code> if source and target contents are identical.
+     * @throws IOException if some I/O error occurs
+     */
+    public boolean transform(final File inJarFile, final File outJarFile) throws IOException {
+        boolean transformed = false;
+        final File dir = outJarFile.getParentFile();
+        if (!dir.exists()) {
+            if (!dir.mkdirs()) {
+                throw new IOException("Couldn't create directory: " + dir.getAbsolutePath());
             }
         }
-    }
-
-    public static void transformJarFile(final File inJarFile, final File targetDir, final String packagesMappingFile) throws IOException {
-        if (!targetDir.exists()) throw new IOException("Couldn't find directory: " + targetDir.getAbsolutePath());
-        final Transformer t = newTransformer(packagesMappingFile);
+        if (!outJarFile.createNewFile()) {
```

Comment:
One thought for future, it would be convenient to have overwrite option that deletes the outJarFile if it exists already, previous to call to createNewFile.  No need to respond, if you agree we could create an issue for that.  When I have used Batava or Eclipse Transformer to run the same script repeatedly, it is a pain to have to ensure the output doesn't exist.

Reply:
Go ahead @scottmarlow  and create issue for that please. I will address this issue in next iteration this week.

Reply:
https://github.com/wildfly-extras/batavia/issues/64